packages/search-adapter/src/empathy/__tests__/response-mappers/empathy-facet-mapper.spec.ts

```diff
@@ -107,21 +117,25 @@ function expectFacetToMatchMock(mappedFacet: HierarchicalFacet | SimpleFacet | N
 function expectBooleanFiltersToHaveSelectedPropertyTo(filters: BooleanFilter[], selected: boolean | null): void {
   filters.forEach(filter => {
     expect(filter.selected).toEqual(selected);
-    if (isHierarchicalFilter(filter) && !!filter.children) {
-      expectBooleanFiltersToHaveSelectedPropertyTo(filter.children, selected);
-    }
   });
 }
 
 function expectAllFiltersToBeValid(filters: Filter[], schema: HierarchicalFilter | SimpleFilter | NumberRangeFilter): void {
   return filters.forEach(filter => {
     expect(filter).toMatchObject(schema);
-    if (isHierarchicalFilter(filter) && filter.children) {
-      expectAllFiltersToBeValid(filter.children, HierarchicalFilterSchema);
-    }
   });
 }
 
-function isHierarchicalFilter(filter: Filter): filter is HierarchicalFilter {
-  return filter.modelName === 'HierarchicalFilter';
+function countHierarchicalChildren(filter: EmpathyFacet | EmpathyFilter): number {
```

Comment:
This is not a good testing practice. Tests should remain simple, it is preferable to hardcode the number of filters rather that to have a recursive function that counts them. Who is asserting that this function is counting them properly?

The only good reason to automate this cont is to avoid changing it if the mock is changed, but I find quite helpful that tests fail if I change the mock they use.

Reply:
Right, I'll get rid of it.